### PR TITLE
feat: archived conversation list [WPB-4429]

### PR DESCRIPTION
**Pull request by Garzas**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-4429" title="WPB-4429" target="_blank">StoryWPB-4429</a>  [UI] Create archived conversation tab and archived conversation list
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
    - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
    - [x] contains a reference JIRA issue number like `SQPIT-764`
    - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
    - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

- archived conversation list
- archived conversation list search
- added test for conversation list view model

https://github.com/wireapp/wire-android-reloaded/assets/13151239/f323868c-22a9-4f58-8afa-405a2c784b76

https://github.com/wireapp/wire-android-reloaded/assets/13151239/9732dc2d-d435-461f-91d2-b50597f82a87

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [1106](https://10.10.124.134/job/AR-build-pipeline/1106/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.5.0(7729090)(7729090)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.5.0(7729090)(7729090)-staging-compat.apk)

**Comment by AndroidBob** [comment]
Build [1124](https://10.10.124.134/job/AR-build-pipeline/1124/) **failed**.

File: app/src/main/kotlin/com/wire/android/WireApplication.kt [hunk 1]
```diff
@@ -112,7 +112,7 @@ class WireApplication : Application(), Configuration.Provider {
                     .detectDiskReads()
                     .detectDiskWrites()
                     .penaltyLog()
-                    .penaltyDeath()
+//                    .penaltyDeath()
```
Comment (borichellow) [inline app/src/main/kotlin/com/wire/android/WireApplication.kt]: shouldn't be reverted?

File: app/src/main/kotlin/com/wire/android/ui/home/conversationslist/model/SearchQuery.kt [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ */
+package com.wire.android.ui.home.conversationslist.model
+
+data class SearchQuery(val text: String, val fromArchive: Boolean)
```
Comment (borichellow) [inline app/src/main/kotlin/com/wire/android/ui/home/conversationslist/model/SearchQuery.kt]: I don't like this `boolean`. For now it's enough, but later the app could also have "folders" and "contacts" later. So IMO `enum class` would be preferable here.

File: app/src/main/kotlin/com/wire/android/ui/home/conversationslist/ConversationListViewModel.kt [hunk 3]
```diff
@@ -123,10 +125,18 @@ class ConversationListViewModel @Inject constructor(
...
 
     private val searchQueryFlow = mutableSearchQueryFlow
-        .asStateFlow()
+        .scan(SearchQuery("", false)) { currentSearchQuery, update ->
+            when (update) {
+                is SearchQueryUpdate.UpdateQuery -> currentSearchQuery.copy(text = update.text)
+                is SearchQueryUpdate.UpdateConversationsSource -> currentSearchQuery.copy(
+                    text = "",
+                    fromArchive = update.fromArchive
+                )
```
Comment (borichellow) [inline app/src/main/kotlin/com/wire/android/ui/home/conversationslist/ConversationListViewModel.kt]:
```
is SearchQueryUpdate.UpdateConversationsSource -> { 
    if (currentSearchQuery.fromArchive != update.fromArchive)
        currentSearchQuery.copy(
            text = "",
            fromArchive = update.fromArchive
        )
    else currentSearchQuery
}
```

File: app/src/main/kotlin/com/wire/android/ui/home/conversationslist/model/SearchQuery.kt (hunk 2 above)
Comment (gongracr) [reply app/src/main/kotlin/com/wire/android/ui/home/conversationslist/model/SearchQuery.kt]: I think that Boris has a point. Also implementation with let's say `sourceType` sealed type shouldn't be much more difficult as now we only have Archive and "normal" conversations.

**Review by gongracr — COMMENTED** [review]
Looking good in general. Just a small comment regarding code simplification on the Search query mechanism. Feel free to ignore if the final implementation turns to be much more complex and would compromise the release of the feature.

File: app/src/main/kotlin/com/wire/android/ui/home/conversationslist/model/SearchQuery.kt (hunk 2 above)
Comment (Garzas) [reply app/src/main/kotlin/com/wire/android/ui/home/conversationslist/model/SearchQuery.kt]: I added comment about it but I can implement it right now 😄

**Comment by AndroidBob** [comment]
Build [1137](https://10.10.124.134/job/AR-build-pipeline/1137/) **failed**.

**Review by gongracr — APPROVED** [review]
Much better now 🦾 Thanks for changes!

**Review by borichellow — APPROVED** [review]
great job, thanks :)

**Review by alexandreferris — APPROVED** [review]

**Comment by AndroidBob** [comment]
Build [1139](https://10.10.124.134/job/AR-build-pipeline/1139/) **failed**.

**Comment by AndroidBob** [comment]
Build [1144](https://10.10.124.134/job/AR-build-pipeline/1144/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.5.0(7745697)(7745697)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.5.0(7745697)(7745697)-staging-compat.apk)

**Pull request merged**